TACC/Core-CMS-Custom
# task/FP-1906 View Registrations Modal

## Overview
Added functionality, updated styling, and isolated HTML for view registration modal
…

## Related


- [FP-1906](https://jira.tacc.utexas.edu/browse/FP-1906)
…

## Changes
- Isolated HTML and styles for view registration modal
- Added functionality (util functions, context variables) to app view for modal
…

## Testing

1. In `src/apps/admin_regis_table/views.py`, replace lines 20-22 with the following snippet:
     <details><summary>Snippet</summary>
     
          import datetime
          registrations_content = [
            (
                1,                                  #registration_id
                datetime.date(2022, 8, 3),          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #file_me
                True,                               #file_pv
                True,                               #file_mc
                True,                               #file_pc
                False,                              #file_dc
                True,                               #submitting_for_self
                'SFTP',                             #submission_method
                'active',                           #registration_status
                'insurance carrier',                #org_type
                'Golden Rule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            ),
            (
                1,                                  #registration_id
                datetime.date(2022, 8, 3),          #posted_date
                12023,                              #applicable_period_start
                122023,                             #applicable_period_end
                True,                               #file_me
                True,                               #file_pv
                True,                               #file_mc
                True,                               #file_pc
                False,                              #file_dc
                True,                               #submitting_for_self
                'SFTP',                             #submission_method
                'active',                           #registration_status
                'insurance carrier',                #org_type
                'Golden Rule Insurance Company',    #business_name
                '7440 Woodland Drive',              #mail_address
                'Indianpolis',                      #city
                'IN',                               #state
                '46278     '                        #zip
            )
        ]
        registrations_entities = [
            (
                5, 1, 5, 46, 11111, 0, 5, 'Garretts Test Business 2', '11-0001111'
            ),
            (
                5, 1, 50000, 47, 1010, 1101, 1, 'A Second Test Entity', '00-0000000'
            )
        ]
        registrations_contacts = [
            (
                52,
                1,
                False,
                'Test Role',
                'Garrett Test Tester',
                '2222222222',
                'notarealemail@emailplace.email'
            ),
            (
                53,
                1,
                False,
                'A 2nd Test Role',
                'Test Garrett 2 Test',
                '5555555555',
                'testemail@testemail.emailtest'
            ),
            (
                54,
                1,
                True,
                'A 3rd and final test role',
                'Test 3rd Role Name Garrett',
                '0000000000',
                None
            )
          ]
        
    </details>
2. Open http://localhost:8000/administration/list-registration-requests/
3. Click on the 'View' button in the last column of either row
4. Confirm the modal content looks like the video below
5. Confirm the modal only closes via 'Close' button in the modal footer




## UI
[Screen Recording 2022-11-09 at 3.03.52 PM.zip](https://github.com/TACC/Core-CMS-Custom/files/9975098/Screen.Recording.2022-11-09.at.3.03.52.PM.zip)

…
